src/main/java/de/themorpheus/edu/taskservice/database/model/TaskModel.java

```diff
@@ -46,6 +48,54 @@
 	@ManyToOne
 	private DifficultyModel difficultyId;
 
+	public TaskModel updateTask(String task) {
+		if (Validation.validateNotNullOrEmpty(task) && !task.equals(this.task)) this.task = task;
+
+		return this;
+	}
+
+	public TaskModel updateDescription(String description) {
+		if (Validation.validateNotNullOrEmpty(description) && !description.equals(this.description))
+			this.description = description;
+
+		return this;
+	}
+
+	public TaskModel updateNecessaryPoints(int necessaryPoints) {
+		if (Validation.greaterOne(necessaryPoints) && necessaryPoints != this.necessaryPoints)
+			this.necessaryPoints = necessaryPoints;
+
+		return this;
+	}
+
+	public TaskModel updateLanguage(String language) {
+		if (Validation.validateNotNullOrEmpty(language) && !language.equals(this.language))
+			this.language = language;
+
+		return this;
+	}
+
+	public TaskModel updateLecture(ControllerResult<LectureModel> lectureResult) {
+		if (lectureResult.isResultPresent() && lectureResult.getResult().equals(this.lectureId))
```

Comment:
Would this ever work?!

Reply:
Why not?

Reply:
You're checking whther the lecture you're getting is the same lecture in the task model, so this would never update

Reply:
ops